packages/flutter_tools/lib/src/globals.dart

```diff
@@ -285,3 +286,12 @@ const String kDefaultFrameworkChannel = 'master';
 
 // Used to build RegExp instances which can detect the VM service message.
 final RegExp kVMServiceMessageRegExp = RegExp(r'The Dart VM service is listening on ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
+
+String? _javaPath;
+
+String? get javaHome => _javaPath ??= findJavaHome(
```

Comment:
This is going to lead to a lot more implicit context accesses, does this need to be global?

Reply:
Or, can't we just have whatever needs to access this directly invoke `findJavaHome`?